lib/Common/DataStructures/BaseDictionary.h

```diff
@@ -860,9 +860,14 @@ namespace JsUtil
             int initSize = max(capacity, 4);
             uint initBucketCount = SizePolicy::GetBucketSize(initSize);
             AssertMsg(initBucketCount > 0, "Size returned by policy should be greater than 0");
-            Allocate(&buckets, &entries, initBucketCount, initSize);
 
-            // Allocation can throw - assign the size only after allocation has succeeded.
+            int* newBuckets = nullptr;
+            EntryType* newEntries = nullptr;
+            Allocate(&newBuckets, &newEntries, initBucketCount, initSize);
+
+            // Allocation can throw - assign only after allocation has succeeded.
+            SetArray<TAllocator>(this->buckets, newBuckets, initBucketCount);
```

Comment:
Why is `this->entries = newEntries` not fine? this->entries and this->buckets are both pointer types- that is, setting them goes through the barrier on the Dictionary instance itself, not on the entries. If you assign `this->entries = blah`, you don't need to see all the bits corresponding to the range of memory symbolized by the entries array, you need to only set the bit corresponding to the dictionary itself, since that's the memory that has changed. That being said, you'll need to do what SetArray does when you do something like memcpy or set individual elements in the array.